### ISSUE TITLE
Firebase config variables are not available when deploying functions

### ISSUE THREAD

**Issue by PascalPrecht** [machinelabs/machinelabs#771]
#### Bug, feature request, or proposal:

I'm experiencing a bug in which I can't deploy the firebase functions due to firebase config variables not being available. This might be a problem with unsupported package versions. See below for more information.

#### What is the expected behavior?

```
$ npm run deploy
```

From within `firebase/functions` should run successfully

#### What is the current behavior?

When running the deploy script, firebase errors out with the error:

```
Error: Error occurred while parsing your function triggers. Please ensure you have the latest firebase-functions SDK by running "npm i --save firebase-functions@latest" inside your functions folder.

Error: Firebase config variables are not available. Please use the latest version of the Firebase CLI to deploy this function.
    at init (/Users/pascalprecht/projects/machinelabs/machinelabs/firebase/functions/dist/node_modules/firebase-functions/lib/config.js:51:15)
    at Object.config (/Users/pascalprecht/projects/machinelabs/machinelabs/firebase/functions/dist/node_modules/firebase-functions/lib/config.js:29:9)
    at Object.<anonymous> (/Users/pascalprecht/projects/machinelabs/machinelabs/firebase/functions/dist/invocation-processor/database.js:5:42)
    at Module._compile (module.js:662:30)
    at Object.Module._extensions..js (module.js:673:10)
    at Module.load (module.js:575:32)
    at tryModuleLoad (module.js:515:12)
    at Function.Module._load (module.js:507:3)
    at Module.require (module.js:606:17)
    at require (internal/module.js:11:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @machinelabs/firebase-functions@0.28.1 deploy: `npm run build && firebase deploy && npm run clean_dist`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @machinelabs/firebase-functions@0.28.1 deploy script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pascalprecht/.npm/_logs/2018-03-24T10_04_32_198Z-debug.log
```

This is very likely caused by those lines of code:

```
const serviceAccount = functions.config().fb_service_account;

if (serviceAccount) {
  conf.credential = admin.credential.cert(<any>{
    private_key: serviceAccount.private_key,
    client_email: serviceAccount.client_email
  });
}
```

So I went ahead and double checked my config variables with

```
$ firebase functions:config:get
```

which outputs my configured `client_email` and `private_key`. Both are correct.

#### Which version of MachineLabs (or the package) is affected ?

`@machinelabs/firebase-functions@27d92f19`

#### Is there anything else we should know?

I digged a bit further and ran into https://github.com/firebase/firebase-tools/issues/609#issuecomment-359042237, which suggests that `firebase-tools` and `firebase-functions` SDK need to be both on latest. 

Which makes me think a possible reason for this issue is that I haven't done any firebase deployments with my current setup so far (new laptop, and we didn't introduce any changes to functions/rules etc), however, I'm running on `firebase-tools@3.17.7` (globally), which might not be compatible anymore with `firebase-functions@0.5.9`. That's the version that currently gets installed.

I will have to try older versions of `firebase-tools` to verify whether this is causing this or not.

**Sidenote**: On that note, we might want to consider switching to `npx` for firebase-tools as well?

Anyways, would be great if anyone can try it out by installing the latest firebase-tools and verifying whether this is reproducable or not.

**Comment by PascalPrecht** [comment]
Okay, I can confirm that this is not a problem when downgrading `firebase-tools` to 3.16.0. 

This means there was probably a breaking change 3.17.x. I'll investigate further and try to figure out what implications it has and what changes are needed if we'd wanna be on `firebase-tools@latest`.

The other option would be to agree on 3.16.x but I'd rather vote for executing `firebase` through `npx` as well and nail the version down in very package.

Opinions welcome!

**Comment by PascalPrecht** [comment]
Alright, so we can get back to a non-breaking environment when upgrade `firebase-functions` to latest but also `firebase-admin`.

`firebase-tools@3.17.x` requires `firebase-functions@0.9.x` which requires `firebase-admin@5.11.x`.

So as mentioned earlier, we can either now go ahead and update `firebase-functions` and `firebase-admin` but this probably breaks the current environment for anyone who doesn't have `firebase-tools@3.17.x` installed. Or, we use `npx` and add `firebase-tools` as an explicit dependency of every package.

Another thing we can do is agreeing on `firebase-tools@3.16.0` installed globally, which I'm not a big fan of. I can go ahead an create a PR for the npx option if everybody agrees.

**Comment by SamVerschueren** [comment]
I'm definitely more fan of `npx` to be honest. We should get rid of all globally installed packages if possible to make sure we all run the same versions. The more contributors we get over time, the more likely it is that people have other (older) versions installed because they use that version at their work for example. That is exactly what happened with `yarn`. At work, we used another version of `yarn` then I used here for MachineLabs. That's pretty annoying so I'm glad we got rid of that and agreed on using `npx`.
